### ISSUE TITLE
Single-line if with no braces and "check-else"

### ISSUE THREAD

**Issue by limitedmage** [palantir/tslint#400]
"check-else" gives an error with this case:

``` typescript
if (condition1)
    doWork();
else
    doOtherWork();
```

There should a way to specify to TSLint that this is a valid coding convention.

**Comment by adidahiya** [comment]
it looks like you want the `one-line` rule turned off since you're not using braces. you can do that with inline comments:

```ts
/* tslint:disable:one-line */
if (condition1)
    doWork();
else
    doOtherWork();
/* tslint:enable:one-line */
```

**Comment by limitedmage** [comment]
The coding conventions my team uses allow omitting braces if the statements are only one line. We don't want to have to put tslint comments every time we do this. We also want to check that, if we are using braces, the else is in the same line as the closing brace, which is what the "check-else" rule does.

**Comment by adidahiya** [comment]
Ok, that makes sense. I'd suggest making a custom rule for your project (tslint is pluggable, see the [readme](https://github.com/palantir/tslint#custom-rules)) for now as I'm not sure when we'll be able to get around to this.

Thoughts on including / excluding this from core rules? @ashwinr @gscshoyru @leeavital

**Comment by ashwinr** [comment]
Agree with Adi's assessment that it's better as a pluggable rule.

Sent from a phone

On Jun 22, 2015, at 12:03, Adi Dahiya <notifications@github.com<mailto:notifications@github.com>> wrote:


Ok, that makes sense. I'd suggest making a custom rule for your project (tslint is pluggable, see the readme<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_tslint-23custom-2Drules&d=BQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=jafYnmwAIy7b95Z05TKJr1hFAxZozsv6_JDDlWyuJk8&m=4P-HbbE6CZdrIuqc2jfB1P7e-aRZOCGbw3-nHU76eTg&s=PLunhaL_p9FKVn8Kwipv7UFOhOIU_vCEuuo0iOhYuOQ&e=>) for now as I'm not sure when we'll be able to get around to this.

Thoughts on including / excluding this from core rules? @ashwinr<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ashwinr&d=BQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=jafYnmwAIy7b95Z05TKJr1hFAxZozsv6_JDDlWyuJk8&m=4P-HbbE6CZdrIuqc2jfB1P7e-aRZOCGbw3-nHU76eTg&s=DhGqrzn6-A2i-D1S5Uu0BPqV-QLHGz6WqWLlL8wt_38&e=> @gscshoyru<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_gscshoyru&d=BQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=jafYnmwAIy7b95Z05TKJr1hFAxZozsv6_JDDlWyuJk8&m=4P-HbbE6CZdrIuqc2jfB1P7e-aRZOCGbw3-nHU76eTg&s=4Ye_8S6J4cHHY4Iqrfrha-OiZoalmVyCJQT-uouUpqg&e=> @leeavital<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_leeavital&d=BQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=jafYnmwAIy7b95Z05TKJr1hFAxZozsv6_JDDlWyuJk8&m=4P-HbbE6CZdrIuqc2jfB1P7e-aRZOCGbw3-nHU76eTg&s=OJ7jLGyymMCwSmTpVpFvKImY56W11mlpPRsdZbHn35k&e=>
